### PR TITLE
Skip identical packed objects

### PR DESCRIPTION
**Pull request by angelini**
This is a very similar PR to https://github.com/gadget-inc/dateilager/pull/27

We want to skip creating a new object if we receive an update to an object that doesn't update the object at all.

The reason we see this in production is that our diff tool, `fsdiff`, monitors file changes using mod time (along with other tricks). Sometimes that time is updated without actually changing a file. For example: `$ touch file.txt`.

For our packed objects, in our use case this is directories in `node_modules`, we see this behaviour in production. When you modify `package.json` it will update the mod time of many directories even when nothing has changed.

Unlike https://github.com/gadget-inc/dateilager/pull/27, we can't use the index trick here because we don't know the packed objects new hash ahead of time.

### CODE REVIEW

**Comment by angelini** [comment]
Decided to write a new integration test for this, coming up soon.

File: test/client_test.go [hunk 1]
```diff
@@ -138,22 +139,54 @@ func buildDiff(tc util.TestCtx, updates map[string]fsdiff_pb.Update_Action) *fsd
 }
 
 func verifyDir(tc util.TestCtx, dir string, files map[string]expectedFile) {
```
Comment (angelini) [inline test/client_test.go:141]: This didn't used to properly walk the directory and our counting check just worked through happenstance.

It now uses the same walk technique as `fsdiff` which only counts directories if they are empty - this matches DL's internal model.

File: test/client_test.go [hunk 2]
```diff
@@ -661,6 +694,75 @@ func TestUpdateObjectsWithMissingFile(t *testing.T) {
 	})
 }
 
+func TestUpdateAndRebuild(t *testing.T) {
```
Comment (angelini) [inline test/client_test.go:697]: These are quite verbose but they test an entire real world workflow.

Comment (scott-rc) [reply test/client_test.go:697]: Sounds like the best kind of test 😄

**Pull request merged**